modules/cnf-provisioning-real-time-and-low-latency-workloads.adoc

```diff
@@ -325,4 +323,121 @@ spec:
+
+[NOTE]
+====
+The capabilities of the processor might impact the latency and throughput of the high priority workloads.
+====
+
+When you configure a node with a power saving configuration, you must configure high priority workloads with performance configuration at the pod level, which means that the configuration applies to all the cores used by the pod.
+
+By disabling P-states and C-states at the pod level, you can configure high priority workloads for best performance and lowest latency.
+
+.Power saving configurations
+[cols="1,2", options="header"]
+|====
+|Annotation
+|Description
+
+a|[source,terminal]
+----
+annotations:
+`cpu-c-states.crio.io: "enable"`
+`cpu-freq-governor.crio.io: "<governor>"`
+----
+|Provides the best performance for a pod by disabling C-states and specifying the governor type for CPU scaling. The `performance` governor is recommended for high priority workloads.
+|====
+
+
+.Prerequisites
+
+* You enabled C-states and OS-controlled P-states in the BIOS
+
+.Procedure
+
+. Generate a `PerformanceProfile` with `per-pod-power-management` set to `true`:
++
+[source,terminal]
+----
+$ podman run --entrypoint performance-profile-creator -v \
+/must-gather:/must-gather:z registry.redhat.io/openshift4/performance-addon-rhel8-operator:v4.11 \
+--mcp-name=worker-cnf --reserved-cpu-count=20 --rt-kernel=true \
+--split-reserved-cpus-across-numa=false --topology-manager-policy=single-numa-node \
+--must-gather-dir-path /must-gather -power-consumption-mode=low-latency \ <1>
+--per-pod-power-management=true > my-performance-profile.yaml
+----
+<1> The `power-consumption-mode` must be `default` or `low-latency` when the `per-pod-power-management` is set to `true`.
+
++
+.Example PerformanceProfile with perPodPowerManagement
+
+[source,yaml]
+----
+apiVersion: performance.openshift.io/v2
+kind: PerformanceProfile
+metadata:
+     name: performance
+spec:
+    [.....]
+    workloadHints:
+        realTime: true
+        highPowerConsumption: false
+        perPodPowerManagement: true
+----
+
+. Set the default `cpufreq` governor with the kernel cmdline option in the `PerformanceProfile` CR:
++
+[source,yaml]
+----
+apiVersion: performance.openshift.io/v2
+kind: PerformanceProfile
+metadata:
+     name: performance
+spec:
+    [.....]
+    additionalKernelArgs:
+    - cpufreq.default_governor=schedutil <1>
+----
+<1> Using the `schedutil` governor is recommended.
+
+. Set the maximum CPU frequency in the `TunedPerformancePatch` CR:
++
+[source,yaml]
+----
+spec:
+  profile:
+  - data: |
+      [sysfs]
+      /sys/devices/system/cpu/intel_pstate/max_perf_pct = <x> <1>
+----
+<1> The `max_perf_pct` controls the maximum frequency the `cpufreq` driver is allowed to set as a percentage of the maximum supported cpu frequency. This value applies to all CPUs.
```

Comment:
Should we mention that the maximum supported frequency can be found in /sys/devices/system/cpu/cpu0/cpufreq/cpuinfo_max_freq?

Reply:
Yes, good idea. Added a sentence about it:
> You can check the maximum supported frequency in `/sys/devices/system/cpu/cpu0/cpufreq/cpuinfo_max_freq`.